### PR TITLE
Adding Metrics API on '/metrics'

### PR DESCRIPTION
**Pull request by tburtchell**

### CODE REVIEW

**Comment by plorenz** [comment]
Unless you prefer not to, can I request that you either squash when merging or squash before merging just to keep the history a bit tidier? It's a lot of commits :)

File: controller/api_impl/metrics-api.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+	return metricsApiHandler, nil
+}
+
+func NewMetricsApiHandler(n *network.Network, options map[interface{}]interface{}) (*MetricsApiHandler, error) {
+	metricsApi := &MetricsApiHandler{
+		options:    options,
+		network:    n,
+		inspectMgr: network.NewInspectionsManager(n),
+	}
+
```
Comment (plorenz) [inline controller/api_impl/metrics-api.go:77]: I don't think we do inline certs anywhere else. Should this just be a path to cert file?

File: controller/api_impl/metrics-api.go [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+func (metricsApi *MetricsApiHandler) IsHandler(r *http.Request) bool {
+	return strings.HasPrefix(r.URL.Path, metricsApi.RootPath())
+}
+
+func (metricsApi *MetricsApiHandler) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
+	metricsApi.handler.ServeHTTP(writer, request)
+}
+
+func (metricsApi *MetricsApiHandler) newHandler() http.Handler {
```
Comment (plorenz) [inline controller/api_impl/metrics-api.go:144]: For consistency, no blank lines at the start of functions, please

File: controller/api_impl/metrics-api.go (hunk 1 above)
Comment (tburtchell) [reply controller/api_impl/metrics-api.go:77]: certainly could be.  I'll look at making that change.

**Review by plorenz — APPROVED** [review]

**Pull request merged**